org.testeditor.web.dropwizard.xtext/src/main/java/org/testeditor/web/dropwizard/xtext/validation/ValidationMarkerMap.xtend

```diff
@@ -0,0 +1,64 @@
+package org.testeditor.web.dropwizard.xtext.validation
+
+import com.google.common.cache.CacheBuilder
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.atomic.AtomicLong
+import javax.inject.Singleton
+import org.slf4j.LoggerFactory
+
+import static java.util.concurrent.TimeUnit.*
+
+@Singleton
+class ValidationMarkerMap {
+	static val logger = LoggerFactory.getLogger(ValidationMarkerUpdater)
+
+	val validationMarkers = new ConcurrentHashMap<String, ValidationSummary>()
+	val lastUpdated = new AtomicLong(System.currentTimeMillis)
+	var futureUpdate = new CompletableFuture<Iterable<ValidationSummary>>()
+	
+	def long getLastUpdated() {
+		return lastUpdated.get
+	}
+
+	def void updateMarkers(Iterable<ValidationSummary> summaries) {
+		if (summaries !== null && summaries.length > 0) {
```

Comment:
null check necessary? length > 0 vs. !empty

Reply:
I'm assuming you're suggesting that, if someone _does_ pass in `null`, its preferable that the method throws an NPE, rather than silently ignoring it…? I guess I could agree with that, although it makes the code a bit less "explicit", which makes me a bit uneasy…